src/promptflow/promptflow/contracts/flow.py

```diff
@@ -697,7 +698,7 @@ def _update_working_dir(cls, working_dir: Path):
     @classmethod
     def from_yaml(cls, flow_file: Path, working_dir=None) -> "Flow":
         """Load flow from yaml file."""
-        working_dir = cls._parse_working_dir(flow_file, working_dir)
+        flow_file, working_dir = cls._parse_working_dir(flow_file, working_dir)
```

Comment:
I guess we can have `flow_file = flow_file.resolve()` when working_dir is not None

Reply:
It would work! You mean, right before original L#701 `with open(working_dir / flow_file, (...), flow_file should be resolved when working_dir is not None, right?